source/slang/slang-check-decl.cpp

```diff
@@ -984,25 +992,10 @@ namespace Slang
         base = TranslateTypeNode(base);
         inheritanceDecl->base = base;
 
-        // For now we only allow inheritance from interfaces, so
-        // we will validate that the type expression names an interface
-
-        if(auto declRefType = as<DeclRefType>(base.type))
-        {
-            if(auto interfaceDeclRef = declRefType->declRef.as<InterfaceDecl>())
-            {
-                return;
-            }
-        }
-        else if(base.type.is<ErrorType>())
-        {
-            // If an error was already produced, don't emit a cascading error.
-            return;
-        }
-
-        // If type expression didn't name an interface, we'll emit an error here
-        // TODO: deal with the case of an error in the type expression (don't cascade)
-        getSink()->diagnose( base.exp, Diagnostics::expectedAnInterfaceGot, base.type);
+        // Note: we do not check whether the type being inherited from
```

Comment:
Correct. The comment is trying to convey that at the point where we check the `InheritanceDecl` itself (which corresponds to the `: SomeBase` clause in the source code), we only check the type expression for `SomeBase` and don't evaluate whether it is actually usable as a base type in context.

Checking whether a base type is valid tends to depend on the derived type (because different derived type declarations will have different constraints), so we let them handle all the more refined checks.